src/databricks/sql/client.py

```diff
@@ -458,8 +501,36 @@ def _prepare_inline_parameters(
 
         return rendered_statement, NO_NATIVE_PARAMS
 
+    def _prepare_dbsql_params_from_list(
+        self, params: ListOfParameters
+    ) -> List[DbsqlParameter]:
+        """Return a list of DbsqlParameter objects from the passed list of params"""
+
+        output = []
+        for p in params:
```

Comment:
minor, but I think the cleaner way to handle this is to have a factory method (could be your __init__ or could be a create_from(...)) for DbsqlParameter that can take p as a value or as a DbsqlParameter.  Then you don't really need this method, because getting your list is [DbsqlParameter.create_from(p) for p in params]